File: 2473-max-sum-of-a-pair-with-equal-sum-of-digits/max-sum-of-a-pair-with-equal-sum-of-digits.cpp
class Solution {
public:
    int maximumSum(vector<int>& nums) {
        int prefix[82] = {0};
        int n = nums.size(), res = -1;
        for (int i = 0; i < n; ++i) {
            int num = nums[i], dig = 0;
            while (num) {
                dig += num % 10;
                num /= 10;
            }
            if (prefix[dig]) {
                res = max(res, nums[i] + prefix[dig]);
                prefix[dig] = max(nums[i], prefix[dig]);
            } else {
                prefix[dig] = nums[i];
            }
        }
        return res;
    }
};

File: 2473-max-sum-of-a-pair-with-equal-sum-of-digits/max-sum-of-a-pair-with-equal-sum-of-digits.py
class Solution:
    def maximumSum(self, nums: List[int]) -> int:
        prefix = [0] * 82
        n, res = len(nums), -1
        for i in range(n):
            num = nums[i]
            dig = 0
            while num:
                dig += num % 10
                num //= 10
            if prefix[dig]:
                res = max(res, nums[i]+prefix[dig])
                prefix[dig] = max(nums[i], prefix[dig])
            else:
                prefix[dig] = nums[i]
        return res